### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by epidemia**
Update:
- jackson
- spring framework
- spring security

Refactoring.

Relates to #403 and #404.

### CODE REVIEW

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java [hunk 1]
```diff
@@ -22,234 +16,204 @@
...
+            @ApiResponse(code = 400,
+                    message = "An error occurred!"),
+            @ApiResponse(code = 404,
+                    message = "The specified (storageId/repositoryId/path) does not exist!")})
     @PreAuthorize("hasAuthority('MANAGEMENT_UNDELETE_TRASH')")
     @RequestMapping(value = "{storageId}/{repositoryId}/{path:.+}",
-                    method = RequestMethod.POST,
-                    produces = MediaType.TEXT_PLAIN_VALUE)
+            method = RequestMethod.POST,
+            produces = MediaType.ALL_VALUE)
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java]: Why have you changed this to `MediaType.ALL_VALUE`.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java [hunk 2]
```diff
@@ -341,23 +305,21 @@ private void generateDirectoryListing(Repository repository,
...
-                            @ApiResponse(code = 400, message = "Bad request."),
-                            @ApiResponse(code = 404, message = "The source/destination storageId/repositoryId/path does not exist!") })
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The path was copied successfully."),
+            @ApiResponse(code = 400, message = "Bad request."),
+            @ApiResponse(code = 404, message = "The source/destination storageId/repositoryId/path does not exist!")})
     @PreAuthorize("hasAuthority('ARTIFACTS_COPY')")
-    @RequestMapping(produces = MediaType.TEXT_PLAIN_VALUE,
-                    value = "/copy/{path:.+}",
-                    method = RequestMethod.POST)
+    @RequestMapping(produces = MediaType.ALL_VALUE,
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java]: Why have you changed this to `MediaType.ALL_VALUE`?

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java [hunk 3]
```diff
@@ -202,12 +207,37 @@ public Query createNativeQuery(String sqlString,
         throw new UnsupportedOperationException("createNativeQuery");
     }
 
+    @Override
+    public StoredProcedureQuery createNamedStoredProcedureQuery(String s) {
```
Comment (carlspring) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: Could you please re-indent you changes to follow the [coding convention](https://github.com/strongbox/strongbox/wiki/Coding-Convention)?

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java [hunk 4]
```diff
@@ -202,12 +207,37 @@ public Query createNativeQuery(String sqlString,
...
+        return null;
+    }
+
+    @Override
+    public StoredProcedureQuery createStoredProcedureQuery(String s) {
```
Comment (carlspring) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: I'm assuming these methods are required by some interface... Shouldn't we have a proper implementation for them?

@epidemia , @sbespalov : What do you think?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/config/WebConfig.java [hunk 5]
```diff
@@ -30,31 +23,34 @@
...
-          StrongboxSecurityConfig.class,
-          StorageApiConfig.class,
-          Maven2LayoutProviderConfig.class,
-          NugetLayoutProviderConfig.class,
-          NpmLayoutProviderConfig.class,
-          StorageCoreConfig.class,
-          SecurityConfig.class,
-          ClientConfig.class,
-          CronTasksConfig.class })
+@ComponentScan({"com.carlspring.strongbox.controllers",
```
Comment (carlspring) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/config/WebConfig.java]: Could you please, not change the formatting of the code? Such changes that are just changes to the indentation make it hard to spot what exactly changed from all the noise...

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java (hunk 3 above)
Comment (epidemia) [reply strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: Fixed

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/config/WebConfig.java (hunk 5 above)
Comment (epidemia) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/config/WebConfig.java]: Fixed

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java (hunk 1 above)
Comment (epidemia) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java]: Some tests did not pass. Apparently, this is caused by the update Spring Framework. I have not yet found a solution to the problem.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java [hunk 6]
```diff
@@ -124,15 +119,14 @@ public ResponseEntity delete()
     @PreAuthorize("hasAuthority('MANAGEMENT_UNDELETE_TRASH')")
     @RequestMapping(value = "{storageId}/{repositoryId}/{path:.+}",
                     method = RequestMethod.POST,
-                    produces = MediaType.TEXT_PLAIN_VALUE)
+                    produces = MediaType.ALL_VALUE)
```
Comment (fuss86) [inline strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java]: I don't understand the necessity of this change.

https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/MediaType.html

 `MediaType.TEXT_PLAIN_VALUE` is still supported.

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java (hunk 2 above)
Comment (fuss86) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java]: I don't understand the necessity of changes around `TrashController.java` and `MavenArtifactController.java` in this PR. What's the issue with old implementation ?

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java [hunk 7]
```diff
@@ -202,12 +199,42 @@ public Query createNativeQuery(String sqlString,
         throw new UnsupportedOperationException("createNativeQuery");
     }
 
+    @Override
+    public StoredProcedureQuery createNamedStoredProcedureQuery(String s)
+    {
+        return null;
```
Comment (fuss86) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: `return null;`

I believe we should `throw new UnsupportedOperationException` for each new method from the persistence API already not supported by us.

throw new UnsupportedOperationException("createNativeQuery");

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxPersistence.java [hunk 8]
```diff
@@ -86,6 +86,18 @@ public synchronized EntityManagerFactory createContainerEntityManagerFactory(Per
         return new OJPAPartitionedEntityManagerPool(properties);
     }
 
+    @Override
+    public void generateSchema(PersistenceUnitInfo info, Map map)
+    {
+
```
Comment (fuss86) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxPersistence.java]: Same as above.

I believe we should `throw new UnsupportedOperationException` for each new method from the persistence API already not supported by us.

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAPartitionedEntityManagerPool.java [hunk 9]
```diff
@@ -41,6 +37,18 @@ public EntityManager createEntityManager(final Map map)
         return createEntityManager(new OJPAProperties(map));
     }
 
+    @Override
+    public EntityManager createEntityManager(SynchronizationType synchronizationType)
+    {
+        return null;
```
Comment (fuss86) [inline strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAPartitionedEntityManagerPool.java]: Same here. `throw new UnsupportedOperationException`

File: strongbox-data-service/pom.xml [hunk 10]
```diff
@@ -159,6 +159,10 @@
             <groupId>org.springframework</groupId>
             <artifactId>spring-aop</artifactId>
         </dependency>
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
+        </dependency>
```
Comment (sbespalov) [inline strongbox-data-service/pom.xml]: why do we need this dependency?

File: strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java (hunk 7 above)
Comment (sbespalov) [reply strongbox-data-service/src/main/java/com/orientechnologies/orient/object/jpa/OJPAObjectDatabaseTxEntityManager.java]: +1
also why this methods were added?

File: strongbox-parent/pom.xml [hunk 11]
```diff
@@ -396,6 +396,12 @@
                 <groupId>com.orientechnologies</groupId>
                 <artifactId>orientdb-object</artifactId>
                 <version>${version.orientdb}</version>
+                <exclusions>
```
Comment (sbespalov) [inline strongbox-parent/pom.xml]: there is `hibernate` dependency there?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java (hunk 6 above)
Comment (carlspring) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/TrashController.java]: +1. Me neither.

**Comment by steve-todorov** [comment]
Hi guys, what is the status of this issue? Is there any update?

File: strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java (hunk 2 above)
Comment (epidemia) [reply strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/maven/MavenArtifactController.java]: Hi!
Some tests were unsuccessful after updating dependencies.
I get the following error: java.lang.AssertionError: 1 expectation failed.
Expected status code <200> but was <406>.

I tried to quickly understand the causes of the problem, but so far I have not found a solution.
For this reason, I changed MediaType for the two methods.

File: strongbox-parent/pom.xml (hunk 11 above)
Comment (epidemia) [reply strongbox-parent/pom.xml]: Yes. See details here: http://mavenrepository.com/artifact/com.orientechnologies/orientdb-object/2.2.18

Comment (fuss86) [reply strongbox-parent/pom.xml]: confirmed :) thanks !

File: strongbox-data-service/pom.xml (hunk 10 above)
Comment (epidemia) [reply strongbox-data-service/pom.xml]: After updating the libraries, I got a compilation error: NoSuchMethodError: javax.persistence.PersistenceContext.synchronization. 
The class PersistenceContext is present in two implementations: javaee-api and hiberbate-jpa-api.
javaee-api has method synchronization, hiberbate-jpa-api has't. So I added this dependency, and excluded the other.

**Comment by carlspring** [comment]
Thanks a lot for your great help! :)

**Pull request merged**